design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,146 @@
+    "signature": "CZgBa6SwfumtC90QoIIh_vfeXgO_I1SfFYrl4RMiCrW_NtMPOoBIbV6WKze1_P6EjnGAsOHoZmEJ3g_7GpTAAuFU6arj2Ku4JDUup--jPTvfAcPxZViQBISTTmwLUXlDqPsIFbG9lfLV-cCM0v_6BbU0dgWba-w-k41PGHQupioSe_kr5b6izo9prSYlCpn_s9GZz41AYDPxiEPZHk-Xe2fjJCRt5vUI_QJKGA8pScMUozFIxXeGeH3nM28W54rqdRQoJxvgW7YUH_5x0nNurLgtJQSZHI38x9uTassLiRluj5AtkB6DCPCvRQiNdq2yyAQOZuVe7Cj8baHYDDrIpJNUjW2VeJ22f-vcGrsoB6X3H7Y2hfzBxw1NQjEaDaz5"
+}
+``` 
+
+In order to work with the server the user **must** encode the response:
+```sh
+echo "Get Conjur access token using an Azure access token"
+# Get an authn-azure Conjur access token for host azure-apps/test-app
+authn_azure_response=$(curl -k -X POST \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  --data "jwt=$azure_access_token" \
+  https://"$CONJUR_SERVER_DNS":8443/authn-azure/test/cucumber/host%2Fazure-apps%2Ftest-app/authenticate)
+authn_azure_access_token=$(echo -n "$authn_azure_response" | base64 | tr -d '\r\n')
+
+echo "Retrieve a secret using the Conjur access token"
+# Retrieve a Conjur secret using the authn-azure Conjur access token
+secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
+  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
+
+echo "Retrieved secret ${secret} from Conjur!!!"
+```
+Taken from: https://github.com/cyberark/conjur/blob/application_identity_validation/ci/authn-azure/run-authn-azure.sh#L16
+
+
+The required situation is to support both `application/json` response (as above) and encoded one.
+The user will have the option to receive an encoded response which skips the encoding process afterwards. 
+
+## Issue description
+1. HTTP server best practices: Using `application\json` as the content type to encoded output (not a JSON) is bad practice.
+    So encoded output shouldn't use `application/json` unless it will be in JSON format. 
+    The `../autheticate` response is in JSON format so it makes sense to use `application\json` as the response content type.
+        As well as `../logic`,  which returns the api key as `text\plain`.
+
+## Solution - Accept HTTP Header
+According to [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept):
+ > The `Accept` request HTTP header advertises which content types the client is able to understand. Using content negotiation, the server then selects one of the proposals, uses it and informs the client of its choice with the `Content-Type` response header.
+
+To conclude, we can choose the response's `Content-Type` by the request's `Accept` header.
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the `Accept` header is `text/plain`. 
+Otherwise, returns decoded json access token as `application/json`.
+
+```ruby
+def authenticate
+    # get the authn access token
+    content_type = (request.headers["Accept"] == "text\plain") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
```

Comment:
Can be removed from design doc - since it might change

Reply:
PSEUDO code